scheduler/src/test/java/io/camunda/zeebe/scheduler/retry/RetryStrategyTest.java

```diff
@@ -116,6 +118,25 @@ public void shouldAbortOnOtherException() {
     assertThat(resultFuture.getException()).isExactlyInstanceOf(RuntimeException.class);
   }
 
+  @Test
+  public void shouldNotInterleaveRetry() {
```

Comment:
:wrench:  Do we want to prevent interleaving the retries, and ensure that the ordering is preserved? Or just that two concurrently submitted tasks can be completed successfully even if they have to be retried? The test verifies the second behavior, which is acceptable. But just rename the method to reflect this.